File: python/567. Permutation in String.cpp
class Solution:
    def is_ok(self, a, b):
        for idx in range(0, 26):
            if a[idx] != b[idx]:
                return False
        return True
    def checkInclusion(self, s1: str, s2: str) -> bool:
        freq1 = [0] * 26
        freq2 = [0] * 26
        for ch in s1:
            idx = ord(ch) - ord('a')
            freq1[idx] = freq1[idx] + 1
        
        low = 0
        sz = len(s1)
        ans = False
        for i, ch in enumerate(s2):
            idx = ord(ch) - ord('a')
            freq2[ idx ] = freq2[idx] + 1
            if i - low + 1 == sz:
                ans = self.is_ok(freq1, freq2)
                idx = ord(s2[low]) - ord('a')
                freq2[idx] = freq2[idx] - 1
                low = low + 1
            
            if ans:
                break
        
        return ans
            
        

File: javascript/567. Permutation in String.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {boolean}
 */

var isOk = function(a, b) {
    for( let i = 0 ; i < 26 ; i++ ) {
        
        if( a[i] != b[i] ) {
            return false;
        }
    }
    return true;
}
var checkInclusion = function(s1, s2) {
    let freq1 = new Array(26).fill(0);
    let freq2 = new Array(26).fill(0);
    let sz = s1.length;
    for( var i = 0 ; i < sz ; i++ ) {
        
        let idx = s1.charCodeAt(i)- 97;
        freq1[idx]++;
    }
    
   
    let ans = false;
    let low = 0;
    for( let i = 0 ; i < s2.length && !ans; i++ ) {
        let idx = s2.charCodeAt(i)- 97;
        freq2[idx]++;
        if( i - low + 1 == sz ) {
            ans = isOk(freq1, freq2);
            idx = s2.charCodeAt(low)- 97;
            freq2[idx]--;
            low++;
        }
        
    }
    
    return ans;
    
    
};
